wire-library/wire-grpc-client/src/jvmMain/kotlin/com/squareup/wire/GrpcClient.kt

```diff
@@ -33,27 +34,48 @@ class GrpcClient private constructor(
   internal val baseUrl: HttpUrl
 ) {
   fun <T : Service> create(service: KClass<T>): T {
-    val methodToService: Map<Method, GrpcMethod<*, *>> =
-        service.java.methods.associate { method -> method to method.toGrpc<Any, Any>() }
+    val methodToService: Map<Method, () -> Any> =
+        service.java.methods.associate { method -> method to method.toGrpc() }
 
     return Proxy.newProxyInstance(
         service.java.classLoader,
         arrayOf<Class<*>>(service.java),
         object : InvocationHandler {
           override fun invoke(proxy: Any, method: Method, args: Array<Any>?): Any {
-            val self: InvocationHandler = this
-
             if (method.declaringClass == Object::class.java) {
-              return method.invoke(self, *(args ?: emptyArray()))
+              return method.invoke(this, *(args ?: emptyArray()))
             }
-
-            val grpcMethod = methodToService[method] as GrpcMethod<*, *>
-            return grpcMethod.invoke(this@GrpcClient)
+            return methodToService.getValue(method).invoke()
           }
         }
     ) as T
   }
 
+  fun <S : Any, R : Any> newCall(method: GrpcMethod<S, R>): GrpcCall<S, R> {
+    return RealGrpcCall(this, method)
+  }
+
+  fun <S : Any, R : Any> newStreamingCall(method: GrpcMethod<S, R>): GrpcStreamingCall<S, R> {
```

Comment:
These two methods are meant for the common version of this type. Just didn't move them there because there's no impl on the other platforms yet.

Reply:
In this kind of scenario, is it an option to move it to common and `actual stuff { TODO() }` on the to-be implemented side?

Reply:
Oh yeah absolutely. I just figured we'd be porting JVM to codegen first. We can do this to unblock other MPP work if we need to though.

Reply:
Was just asking for now, don't think it blocks anyone yet. 👍